modules/stvBidAdapter.js

```diff
@@ -0,0 +1,308 @@
+import { deepAccess } from '../src/utils.js';
+import { config } from '../src/config.js';
+import { registerBidder } from '../src/adapters/bidderFactory.js';
+import { BANNER, VIDEO } from '../src/mediaTypes.js';
+import {includes} from '../src/polyfill.js';
+
+const BIDDER_CODE = 'stv';
+const ENDPOINT_URL = 'https://ads.smartstream.tv/r/';
+const ENDPOINT_URL_DEV = 'https://ads.smartstream.tv/r/';
+const GVLID = 134;
+const VIDEO_ORTB_PARAMS = {
+  'minduration': 'min_duration',
+  'maxduration': 'max_duration',
+  'maxbitrate': 'max_bitrate',
+  'api': 'api',
+};
+
+export const spec = {
+  code: BIDDER_CODE,
+  gvlid: GVLID,
+  aliases: [],
+  supportedMediaTypes: [BANNER, VIDEO],
+  isBidRequestValid: function(bid) {
+    return !!(bid.params.placement);
+  },
+  buildRequests: function(validBidRequests, bidderRequest) {
+    return validBidRequests.map(bidRequest => {
+      const params = bidRequest.params;
+
+      const placementId = params.placement;
+      const rnd = Math.floor(Math.random() * 99999999999);
+      const referrer = bidderRequest.refererInfo.page;
+      const bidId = bidRequest.bidId;
+      const isDev = params.devMode || false;
+      const pbcode = bidRequest.adUnitCode || false; // div id
+      // const auctionId = bidRequest.auctionId || false;
+
+      let endpoint = isDev ? ENDPOINT_URL_DEV : ENDPOINT_URL;
+
+      let mediaTypesInfo = getMediaTypesInfo(bidRequest);
+      let type = isBannerRequest(bidRequest) ? BANNER : VIDEO;
+      let sizes = mediaTypesInfo[type];
+
+      let payload = {};
+      if (isVideoRequest(bidRequest)) {
+        payload = {
+          _f: 'vast2',
+          alternative: 'prebid_js',
+          _ps: placementId,
+          srw: sizes ? sizes[0].width : 0,
+          srh: sizes ? sizes[0].height : 0,
+          idt: 100,
+          rnd: rnd,
+          ref: referrer,
+          bid_id: bidId,
+          pbver: '$prebid.version$',
+        };
+      } else {
+        payload = {
+          _f: 'html',
+          alternative: 'prebid_js',
+          _ps: placementId,
+          srw: sizes ? sizes[0].width : 0,
+          srh: sizes ? sizes[0].height : 0,
+          idt: 100,
+          rnd: rnd,
+          ref: referrer,
+          bid_id: bidId,
+          pbver: '$prebid.version$'
+        };
```

Comment:
seems like all properties other than "_f" are shared in these two conditional blocks. define the object once and adjust the property value within each block